File: LES_EXAMENS/EXAM_PYTHON_ET_EXAM_ALGO_C__2025_2026_IA_RATTRAPAGE/exerciceB.py
while True:
    n=int(input("saisir un entier n>0: "))
    if n>0:
        break
X=float(input("saisir un reel X: "))
S=0
for i in range(1,n+1):
    S=S+i*(X**i)
print(f"la somme S={S}")

File: LES_EXAMENS/EXAM_PYTHON_ET_EXAM_ALGO_C__2025_2026_IA_RATTRAPAGE/exerciceB.c
#include <stdio.h>
#include<math.h>
int n;
float X,S;
int main(){
    do{
        printf("saisir un entier n>0: ");
        scanf("%d",&n);
    }while(n<=0);
    printf("saisir un reel X: ");
    scanf("%f",&X);
    S=0;
    for(int i=1;i<=n;i++){
        S=S+i*(pow(X, i));
    }
    printf("la somme S=%f",S);
    return 0;
}